test/integration-tests/V3ToV4Migration.test.ts

```diff
@@ -945,4 +961,301 @@ describe('V3 to V4 Migration Tests:', () => {
+      await wethContract.connect(bob).approve(v3NFTPositionManager.address, MAX_UINT)
+
+      // mint the nft to bob on v3
+      const tx = await v3NFTPositionManager.mint({
+        token0: USDC.address,
+        token1: WETH.address,
+        fee: FeeAmount.LOW,
+        tickLower: 0,
+        tickUpper: 194980,
+        amount0Desired: expandTo6DecimalsBN(2500),
+        amount1Desired: expandTo18DecimalsBN(1),
+        amount0Min: 0,
+        amount1Min: 0,
+        recipient: bob.address,
+        deadline: MAX_UINT,
+      })
+
+      const receipt = await tx.wait()
+      const transferEvent = receipt.events?.find((event) => event.event === 'IncreaseLiquidity')
+      let tokenIdv3 = transferEvent?.args?.tokenId
+
+      // permit, decrease, collect, burn
+      const encodedErc721PermitCall = await permit()
+      const encodedDecreaseCall = await decreaseLiquidity()
+      // set receiver to v4posm
+      const encodedCollectCall = collect()
+      const encodedBurnCall = encodeBurn(tokenIdv3)
+
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_PERMIT, [encodedErc721PermitCall])
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedDecreaseCall])
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedCollectCall])
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedBurnCall])
+
+      const encodedModifyLiquiditiesCallWithIncrease = increaseSettleSweep()
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithIncrease])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // bob successfully sweeped his usdc and weth from the v4 position manager
+      expect(await wethContract.balanceOf(v4PositionManager.address)).to.eq(0)
+      expect(await usdcContract.balanceOf(v4PositionManager.address)).to.eq(0)
+    })
+  })
+
+  function mintSettleSweep(): string {
+    const mintParams = {
+      LiquidityRange: {
+        PoolKey: {
+          currency0: USDC.address,
+          currency1: WETH.address,
+          fee: FeeAmount.LOW,
+          tickSpacing: 10,
+          IHooks: '0x0000000000000000000000000000000000000000',
+        },
+        tickLower: 0,
+        tickUpper: 194980,
+      },
+      liquidity: '6000000',
+      owner: bob.address,
+      hookData: '0x',
+    }
+
+    const encodedMintData = encodeMintData(mintParams)
+
+    const encodedSettleBalance0 = encodeSettleBalance(mintParams.LiquidityRange.PoolKey.currency0, MAX_UINT)
+    const encodedSettleBalance1 = encodeSettleBalance(mintParams.LiquidityRange.PoolKey.currency1, MAX_UINT)
+
+    const encodeSweep0 = encodeSweep(mintParams.LiquidityRange.PoolKey.currency0, bob.address)
+    const encodeSweep1 = encodeSweep(mintParams.LiquidityRange.PoolKey.currency1, bob.address)
+
+    const unlockDataParams = {
+      actions: [MINT_POSITION, SETTLE_WITH_BALANCE, SETTLE_WITH_BALANCE, SWEEP, SWEEP],
```

Comment:
I think it could be great to have a new javascript helper file like planner.ts to do all this for v4?
so that you can do

```
v4Plan.addCommand(MINT_POSITION, [mint params])
```

and it inside that function performs `encode(mintParams)` and then pushes them to an array?

no rush if youd rather focus on finishing this PR before EOW - we can do that as clean up next week